Add more error details for CEA failure

aholappa opened the pull request:
_Result-Code_ was not enough to identify some error cases. In one case, the code is a generic "DIAMETER_UNABLE_TO_COMPLY" and the specific error is in the _Error-Message_ AVP:

```
	Result-Code {Code:268,Flags:0x40,Length:12,VendorId:0,Value:Unsigned32{5012}}
	Error-Message {Code:281,Flags:0x0,Length:40,VendorId:0,Value:UTF8String{Origin-Host already connected},Padding:3}
```

This PR:
- adds _Failed-AVP_ and _Error-Message_ to CEA struct
- exposes CEA in ErrFailedResultCode

fiorix commented on diam/sm/smparser/cea.go:
```diff
@@ -21,13 +21,16 @@ type CEA struct {
 // answer (CEA) contains a Result-Code AVP that is not success (2001).
 type ErrFailedResultCode struct {
 	Code uint32
+	Cea  *CEA
```
We should call this field CEA (not Cea) and remove the Code field due to duplication. I wouldn't mind a `Code()` method in this struct as a shortcut to return `CEA.ResultCode`.

aholappa commented:
- Changed CEA to be an anonymous struct field. This way the fields can be accessed directly via the error struct (e.ResultCode, e.OriginHost etc.)
- Added test case for FailedAVP

fiorix commented:
👍
